Python/Product/Analysis/Intellisense/AnalysisQueue.cs

```diff
@@ -165,24 +167,20 @@ sealed class WaitAnalyzableTask : IAnalyzable {
             }
         }
 
-        public int AnalysisPending {
-            get {
-                return _analysisPending;
-            }
-        }
+        public int AnalysisPending => _analysisPending;
 
         #region IDisposable Members
-
         public void Dispose() {
             Stop();
             _workEvent.Dispose();
             _cancel.Dispose();
+            _disposed = true;
```

Comment:
Should we set the flag first so that items stop being queued while we are stopping?